src/handlers/feature_handlers.cpp

```diff
@@ -4942,7 +4977,590 @@ void vehCarModelFeatureHandler::DrawShadow() {
 
 void vehCarModelFeatureHandler::DrawGlow() {
     auto model = reinterpret_cast<vehCarModel*>(this);
-    model->vehCarModel::DrawGlow();
+    if (!model->GetVisible())
+        return;
+
+    //get our geometry id
+    int geomSetId = model->GetGeomIndex();
+    int geomSetIdOffset = geomSetId - 1;
+
+    //get shaders
+    auto mainGeomEntry = lvlInstance::GetGeomTableEntry(geomSetIdOffset);
+    auto shaders = mainGeomEntry->pShaders[model->getVariant()];
+
+    //get car stuff we use to determine what to darw
+    auto car = model->getCar();
+    auto carsim = model->getCarSim();
+    auto siren = car->getSiren();
+    auto curDamage = car->getCarDamage()->getCurDamage();
+    auto maxDamage = car->getCarDamage()->getMaxDamage();
+    int gear = carsim->getTransmission()->getGear();
+    if (curDamage > maxDamage && vehCarModel::MWStyleTotaledCar)
+        return;
+
+    //setup renderer
+    Matrix44::Convert(gfxRenderState::sm_World, model->getCarMatrix());
+    gfxRenderState::m_Touched = gfxRenderState::m_Touched | 0x88;
+
+    //get signals
+    modStatic* slight0 = lvlInstance::GetGeomTableEntry(geomSetIdOffset + 5)->GetHighestLOD();
+    modStatic* slight1 = lvlInstance::GetGeomTableEntry(geomSetIdOffset + 6)->GetHighestLOD();
+
+    //get taillight signals
+    modStatic* tslight0 = lvlInstance::GetGeomTableEntry(geomSetIdOffset + 101)->GetHighestLOD();
+    modStatic* tslight1 = lvlInstance::GetGeomTableEntry(geomSetIdOffset + 102)->GetHighestLOD();
+
+    //check signal clock
+    bool drawSignal = fmod(datTimeManager::ElapsedTime, 1.f) > 0.5f;
+
+    //draw stuff!
+    if (drawSignal && car->IsPlayer()) {
+        if (vehCarModel::LeftSignalLightState || vehCarModel::HazardLightsState) {
+            if (slight0 != nullptr)
+                slight0->Draw(shaders);
+            if (tslight0 != nullptr)
+                tslight0->Draw(shaders);
+        }
+        if (vehCarModel::RightSignalLightState || vehCarModel::HazardLightsState) {
+            if (slight1 != nullptr)
+                slight1->Draw(shaders);
+            if (tslight1 != nullptr)
+                tslight1->Draw(shaders);
+        }
+    }
+
+    //draw taillight signals for player
+    if (car->IsPlayer()) {
+        if (!vehCarModel::LeftSignalLightState && !vehCarModel::HazardLightsState) {
+            if (tslight0 != nullptr) {
+                //draw brake copy
+                if (carsim->getBrake() > 0.1)
+                    tslight0->Draw(shaders);
+                //draw headlight copy
+                if (vehCarModel::HeadlightsState)
+                    tslight0->Draw(shaders);
```

Comment:
Nah they shouldn't because if they're merged it will draw one of the both lights only while our target is drawing them both together indeed so one when the car brakes and one when the headlights are on. Well, I didn't expect you asking me about this one because these comments explain that already :P